src/devices/duplotrainbasespeedometer.ts

```diff
@@ -15,6 +15,10 @@ export class DuploTrainBaseSpeedometer extends Device {
     }
 
     public receive (message: Buffer) {
+        if (this.hub.autoParse) {
```

Comment:
These are quite repetitive, and error prone when new devices are added (ie. could forget to do this). Could the check be done at hub level, calling either receive or autoParse?

Reply:
I am thinking about another way to do this:

1. Define all known modes like auto parsed ones and add them in the constructor.
2. always use `Device.receive` to parse incoming messages (including mode information) and emit lego named events
3. make the specific device classes to listen on their lego events to emit proper ones

No more `receive` method in specific device classes, just event handlers that could use private methods to handle mode calculation and formating.

Reply:
I totally forgot about the subscription logic in the comment above. It will need the correct mode map.